File: src/main/java/resonantinduction/archaic/blocks/RenderMillstone.java
package resonantinduction.archaic.blocks;

import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.ForgeDirection;
import resonantinduction.core.render.RenderItemOverlayTile;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderMillstone extends RenderItemOverlayTile
{
	private final RenderBlocks renderBlocks = new RenderBlocks();

	@Override
	public void renderTileEntityAt(TileEntity tileEntity, double x, double y, double z, float var8)
	{
		if (tileEntity instanceof TileMillstone)
		{
			TileMillstone tile = (TileMillstone) tileEntity;
			renderItemOnSides(tileEntity, tile.getStackInSlot(0), x, y, z, "Empty");
		}
	}
}

File: src/main/scala/edx/basic/process/RenderMillstone.scala
package edx.basic.process

import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import resonant.lib.render.RenderItemOverlayUtility

@SideOnly(Side.CLIENT)
class RenderMillstone extends TileEntitySpecialRenderer
{
  private final val renderBlocks: RenderBlocks = new RenderBlocks

  def renderTileEntityAt(tileEntity: TileEntity, x: Double, y: Double, z: Double, var8: Float)
  {
    if (tileEntity.isInstanceOf[TileSieve])
    {
      val tile: TileSieve = tileEntity.asInstanceOf[TileSieve]
      RenderItemOverlayUtility.renderItemOnSides(tileEntity, tile.getStackInSlot(0), x, y, z, "")
    }
  }

}